aggregator/follower_flush_mgr.go

```diff
@@ -150,11 +152,12 @@ func (mgr *followerFlushManager) Prepare(buckets []*flushBucket) (flushTask, tim
 		if mgr.flushMode == forcedFollowerFlush && durationSinceLastFlush >= mgr.forcedFlushWindowSize {
 			flushBeforeNanos := now.Add(-mgr.maxBufferSize).UnixNano()
 			flushersByInterval = mgr.flushersFromForcedFlush(buckets, flushBeforeNanos)
+			needsFlush = true
 			mgr.metrics.forcedFlush.Inc(1)
 		}
 	}
 
-	if len(flushersByInterval) == 0 {
+	if !needsFlush {
```

Comment:
is this for readability or is there a case where len(flushersByInterval) == 0 does not equal to !needsFflush?

Reply:
So I changed it so that if the kv is updated, or it's time to do a forced flush, we'll flush and **update the flush time** even if len(flushersByInterval) == 0 (e.g., the instance doesn't own any shards). Otherwise if we don't update the time, the forced flush will be performed every second since `durationSinceLastFlushed` will keep increasing.

Reply:
Got it, maybe add a comment for this?